:rotating_light: NIGHTLY DEPENDENCIES TEST: failure on `refs/heads/main`

The run `10297598102` of the workflow testing Nilearn with the nightly build of its dependencies failed on `refs/heads/main`.

You can view the report here: https://github.com/nilearn/nilearn/actions/runs/10297598102

Pinging @nilearn/core

To reproduce

```
tox -e test_nightly -- nilearn/glm/tests/test_dmtx.py::test_create_second_level_design
```

passed with

matplotlib-3.10.0.dev364+gb485f5a9e1 pandas-3.0.0.dev0+1178.g236d89b856 scikit-learn-1.6.dev0 scipy-1.15.0.dev0 numpy-2.1.0.dev0

started failing with

matplotlib-3.10.0.dev370+g6b7b999274 pandas-3.0.0.dev0+1194.gd966462196 scikit-learn-1.6.dev0 scipy-1.15.0.dev0 numpy-2.1.0.dev0

stacktrace of the test failure: feels like a pandas 3.0 issue

<details>
<pre>
========================================================== FAILURES ===========================================================
_______________________________________________ test_create_second_level_design _______________________________________________

self = NumpyBlock: 1 dtype: int64, indexer = (0,), value = np.float64(0.75)

    def setitem(self, indexer, value) -> Block:
        """
        Attempt self.values[indexer] = value, possibly creating a new array.
    
        Parameters
        ----------
        indexer : tuple, list-like, array-like, slice, int
            The subset of self.values to set
        value : object
            The value being set
    
        Returns
        -------
        Block
    
        Notes
        -----
        `indexer` is a direct slice/positional indexer. `value` must
        be a compatible shape.
        """
    
        value = self._standardize_fill_value(value)
    
        values = cast(np.ndarray, self.values)
        if self.ndim == 2:
            values = values.T
    
        # length checking
        check_setitem_lengths(indexer, value, values)
    
        if self.dtype != _dtype_obj:
            # GH48933: extract_array would convert a pd.Series value to np.ndarray
            value = extract_array(value, extract_numpy=True)
        try:
>           casted = np_can_hold_element(values.dtype, value)

indexer    = (0,)
self       = NumpyBlock: 1 dtype: int64
value      = np.float64(0.75)
values     = array([0])

.tox/test_nightly/lib/python3.12/site-packages/pandas/core/internals/blocks.py:1098: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

dtype = dtype('int64'), element = np.float64(0.75)

    def np_can_hold_element(dtype: np.dtype, element: Any) -> Any:
        """
        Raise if we cannot losslessly set this element into an ndarray with this dtype.
    
        Specifically about places where we disagree with numpy.  i.e. there are
        cases where numpy will raise in doing the setitem that we do not check
        for here, e.g. setting str "X" into a numeric ndarray.
    
        Returns
        -------
        Any
            The element, potentially cast to the dtype.
    
        Raises
        ------
        ValueError : If we cannot losslessly store this element with this dtype.
        """
        if dtype == _dtype_obj:
            return element
    
        tipo = _maybe_infer_dtype_type(element)
    
        if dtype.kind in "iu":
            if isinstance(element, range):
                if _dtype_can_hold_range(element, dtype):
                    return element
                raise LossySetitemError
    
            if is_integer(element) or (is_float(element) and element.is_integer()):
                # e.g. test_setitem_series_int8 if we have a python int 1
                #  tipo may be np.int32, despite the fact that it will fit
                #  in smaller int dtypes.
                info = np.iinfo(dtype)
                if info.min <= element <= info.max:
                    return dtype.type(element)
                raise LossySetitemError
    
            if tipo is not None:
                if tipo.kind not in "iu":
                    if isinstance(element, np.ndarray) and element.dtype.kind == "f":
                        # If all can be losslessly cast to integers, then we can hold them
                        with np.errstate(invalid="ignore"):
                            # We check afterwards if cast was losslessly, so no need to show
                            # the warning
                            casted = element.astype(dtype)
                        comp = casted == element
                        if comp.all():
                            # Return the casted values bc they can be passed to
                            #  np.putmask, whereas the raw values cannot.
                            #  see TestSetitemFloatNDarrayIntoIntegerSeries
                            return casted
                        raise LossySetitemError
    
                    elif isinstance(element, ABCExtensionArray) and isinstance(
                        element.dtype, CategoricalDtype
                    ):
                        # GH#52927 setting Categorical value into non-EA frame
                        # TODO: general-case for EAs?
                        try:
                            casted = element.astype(dtype)
                        except (ValueError, TypeError) as err:
                            raise LossySetitemError from err
                        # Check for cases of either
                        #  a) lossy overflow/rounding or
                        #  b) semantic changes like dt64->int64
                        comp = casted == element
                        if not comp.all():
                            raise LossySetitemError
                        return casted
    
                    # Anything other than integer we cannot hold
>                   raise LossySetitemError
E                   pandas.errors.LossySetitemError

dtype      = dtype('int64')
element    = np.float64(0.75)
tipo       = dtype('float64')

.tox/test_nightly/lib/python3.12/site-packages/pandas/core/dtypes/cast.py:1827: LossySetitemError

During handling of the above exception, another exception occurred:

    def test_create_second_level_design():
        subjects_label = ["02", "01"]  # change order to test right output order
        regressors = [["01", 0.1], ["02", 0.75]]
        regressors = pd.DataFrame(regressors, columns=["subject_label", "f1"])
>       design = make_second_level_design_matrix(subjects_label, regressors)

regressors =   subject_label    f1
0            01  0.10
1            02  0.75
subjects_label = ['02', '01']

nilearn/glm/tests/test_dmtx.py:455: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
nilearn/glm/first_level/design_matrix.py:508: in make_second_level_design_matrix
    design_matrix.loc[ridx, conf_name] = confounds_value
        conf_name  = 'f1'
        confounds  =   subject_label    f1
0            01  0.10
1            02  0.75
        confounds_name = ['f1']
        confounds_value = np.float64(0.75)
        conrow     = 0    False
1     True
Name: subject_label, dtype: bool
        design_columns = ['f1', 'intercept']
        design_matrix =    f1  intercept
0   0          1
        ridx       = 0
        subject_label = '02'
        subjects_label = ['02', '01']
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/indexing.py:918: in __setitem__
    iloc._setitem_with_indexer(indexer, value, self.name)
        iloc       = <pandas.core.indexing._iLocIndexer object at 0x7bc9b53f0aa0>
        indexer    = (0, 0)
        key        = (0, 'f1')
        self       = <pandas.core.indexing._LocIndexer object at 0x7bc9b53c6ee0>
        value      = np.float64(0.75)
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/indexing.py:1936: in _setitem_with_indexer
    self._setitem_with_indexer_split_path(indexer, value, name)
        arr        = array([[0],
       [1]])
        ax         = Index(['f1', 'intercept'], dtype='object')
        i          = 1
        idx        = 0
        indexer    = (0, 0)
        info_axis  = 1
        name       = 'loc'
        nindexer   = [0, 0]
        self       = <pandas.core.indexing._iLocIndexer object at 0x7bc9b53f0aa0>
        take_split_path = True
        val        = np.float64(0.75)
        value      = np.float64(0.75)
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/indexing.py:2029: in _setitem_with_indexer_split_path
    self._setitem_single_column(loc, value, pi)
        ilocs      = [0]
        indexer    = (0, 0)
        info_axis  = 0
        loc        = 0
        lplane_indexer = 1
        name       = 'loc'
        pi         = 0
        self       = <pandas.core.indexing._iLocIndexer object at 0x7bc9b53f0aa0>
        value      = np.float64(0.75)
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/indexing.py:2166: in _setitem_single_column
    self.obj._mgr.column_setitem(loc, plane_indexer, value)
        dtype      = dtype('int64')
        is_full_setter = False
        is_null_setter = False
        loc        = 0
        pi         = 0
        plane_indexer = 0
        self       = <pandas.core.indexing._iLocIndexer object at 0x7bc9b53f0aa0>
        value      = np.float64(0.75)
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/internals/managers.py:1463: in column_setitem
    new_mgr = col_mgr.setitem((idx,), value)
        col_mgr    = SingleBlockManager
Items: Index([0], dtype='int64')
NumpyBlock: 1 dtype: int64
        idx        = 0
        inplace_only = False
        loc        = 0
        self       = BlockManager
Items: Index(['f1', 'intercept'], dtype='object')
Axis 1: Index([0], dtype='int64')
NumpyBlock: slice(0, 2, 1), 2 x 1, dtype: int64
        value      = np.float64(0.75)
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/internals/managers.py:581: in setitem
    return self.apply("setitem", indexer=indexer, value=value)
        indexer    = (0,)
        self       = SingleBlockManager
Items: Index([0], dtype='int64')
NumpyBlock: 1 dtype: int64
        value      = np.float64(0.75)
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/internals/managers.py:438: in apply
    applied = getattr(b, f)(**kwargs)
        align_keys = []
        aligned_args = {}
        b          = NumpyBlock: 1 dtype: int64
        f          = 'setitem'
        kwargs     = {'indexer': (0,), 'value': np.float64(0.75)}
        result_blocks = []
        self       = SingleBlockManager
Items: Index([0], dtype='int64')
NumpyBlock: 1 dtype: int64
.tox/test_nightly/lib/python3.12/site-packages/pandas/core/internals/blocks.py:1101: in setitem
    nb = self.coerce_to_target_dtype(value, raise_on_upcast=True)
        indexer    = (0,)
        self       = NumpyBlock: 1 dtype: int64
        value      = np.float64(0.75)
        values     = array([0])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = NumpyBlock: 1 dtype: int64, other = np.float64(0.75), raise_on_upcast = True

    @final
    def coerce_to_target_dtype(self, other, raise_on_upcast: bool) -> Block:
        """
        coerce the current block to a dtype compat for other
        we will return a block, possibly object, and not raise
    
        we can also safely try to coerce to the same dtype
        and will receive the same block
        """
        new_dtype = find_result_type(self.values.dtype, other)
        if new_dtype == self.dtype:
            # GH#52927 avoid RecursionError
            raise AssertionError(
                "Something has gone wrong, please report a bug at "
                "https://github.com/pandas-dev/pandas/issues"
            )
    
        # In a future version of pandas, the default will be that
        # setting `nan` into an integer series won't raise.
        if (
            is_scalar(other)
            and is_integer_dtype(self.values.dtype)
            and isna(other)
            and other is not NaT
            and not (
                isinstance(other, (np.datetime64, np.timedelta64)) and np.isnat(other)
            )
        ):
            raise_on_upcast = False
        elif (
            isinstance(other, np.ndarray)
            and other.ndim == 1
            and is_integer_dtype(self.values.dtype)
            and is_float_dtype(other.dtype)
            and lib.has_only_ints_or_nan(other)
        ):
            raise_on_upcast = False
    
        if raise_on_upcast:
>           raise TypeError(f"Invalid value '{other}' for dtype '{self.values.dtype}'")
E           TypeError: Invalid value '0.75' for dtype 'int64'

new_dtype  = dtype('float64')
other      = np.float64(0.75)
raise_on_upcast = True
self       = NumpyBlock: 1 dtype: int64

.tox/test_nightly/lib/python3.12/site-packages/pandas/core/internals/blocks.py:469: TypeError
</pre>
</details>

--- a/.github/workflows/nightly_dependencies.yml
+++ b/.github/workflows/nightly_dependencies.yml
@@ -19,6 +19,10 @@ on:
 
     workflow_dispatch:
 
+# Force to use color
+env:
+    FORCE_COLOR: true
+
 concurrency:
     group: ${{ github.workflow }}-${{ github.ref }}
     cancel-in-progress: true

--- a/nilearn/glm/first_level/design_matrix.py
+++ b/nilearn/glm/first_level/design_matrix.py
@@ -488,7 +488,7 @@ def make_second_level_design_matrix(subjects_label, confounds=None):
         raise ValueError("Design matrix columns do not have unique names")
 
     # float dtype necessary for linalg
-    design_matrix = pd.DataFrame(columns=design_columns, dtype=np.float64)
+    design_matrix = pd.DataFrame(columns=design_columns, dtype="float64")
     for ridx, subject_label in enumerate(subjects_label):
         design_matrix.loc[ridx] = [0.0] * len(design_columns)
         design_matrix.loc[ridx, "intercept"] = 1.0

--- a/nilearn/glm/first_level/design_matrix.py
+++ b/nilearn/glm/first_level/design_matrix.py
@@ -488,10 +488,10 @@ def make_second_level_design_matrix(subjects_label, confounds=None):
         raise ValueError("Design matrix columns do not have unique names")
 
     # float dtype necessary for linalg
-    design_matrix = pd.DataFrame(columns=design_columns, dtype=float)
+    design_matrix = pd.DataFrame(columns=design_columns, dtype=np.float64)
     for ridx, subject_label in enumerate(subjects_label):
-        design_matrix.loc[ridx] = [0] * len(design_columns)
-        design_matrix.loc[ridx, "intercept"] = 1
+        design_matrix.loc[ridx] = [0.0] * len(design_columns)
+        design_matrix.loc[ridx, "intercept"] = 1.0
         if confounds is not None:
             conrow = confounds["subject_label"] == subject_label
             if np.sum(conrow) > 1:

--- a/nilearn/glm/tests/test_dmtx.py
+++ b/nilearn/glm/tests/test_dmtx.py
@@ -453,7 +453,7 @@ def test_create_second_level_design():
     regressors = [["01", 0.1], ["02", 0.75]]
     regressors = pd.DataFrame(regressors, columns=["subject_label", "f1"])
     design = make_second_level_design_matrix(subjects_label, regressors)
-    expected_design = np.array([[0.75, 1], [0.1, 1]])
+    expected_design = np.array([[0.75, 1.0], [0.1, 1.0]])
     assert_array_equal(design, expected_design)
     assert len(design.columns) == 2
     assert len(design) == 2

